packages/react-tinacms-editor/src/state/plugins/Common/index.tsx

```diff
@@ -62,5 +63,26 @@ export const commonPlugin = new Plugin({
         return false
       },
     },
+    handleKeyDown(view: EditorView, event: KeyboardEvent) {
+      // If user is at end of current line and there is a code mark an arrow-right will
+      // take user out of the code mark.
+      if (event.key !== 'ArrowRight') return false
+      const { selection, schema } = view.state
+      const { code } = schema.marks
+      const selectionToCodeMark = selection.$to
+        .marks()
+        .some(mark => mark.type === code)
+      if (!selectionToCodeMark) return false
+      const selectionIsAtEnd =
+        selection.$to.node().nodeSize - 2 === selection.$to.parentOffset
+      if (!selectionIsAtEnd) return false
+      const { state, dispatch } = view
+      dispatch(
+        state.tr
+          .insertText(' ', selection.$to.pos)
+          .removeMark(selection.$to.pos, selection.$to.pos + 1, code)
+      )
+      return true
```

Comment:
Awesome! Could you just extract this to it's own function? Maybe something like `exitCodeMarkOnArrowRight`

```ts
handleKeyDown(view, event) {
  return exitCodeMarkOnArrowRight(view, event)
}
```